src/pages/WorkfilesPage/WorkfileDetail.jsx

```diff
@@ -2,12 +2,42 @@ import { Section, Panel } from '@ynput/ayon-react-components'
 import { PathField } from '/src/containers/fieldFormat'
 import Thumbnail from '/src/containers/thumbnail'
 import AttributeTable from '/src/containers/attributeTable'
+import { useState } from 'react'
 import { useSelector } from 'react-redux'
 import { useGetWorkfileByIdQuery } from '/src/services/getWorkfiles'
+import { useGetSiteRootsQuery } from '/src/services/customRoots'
 import { toast } from 'react-toastify'
+import SiteDropdown from '/src/containers/SiteDropdown'
+
+const getCurrentPlatform = () => {
+  const platform = window.navigator.userAgent.toLowerCase()
+
+  if (platform.includes('win')) {
+    return 'windows'
+  } else if (platform.includes('mac')) {
+    return 'darwin'
+  } else if (platform.includes('linux')) {
+    return 'linux'
+  } else {
+    return 'other'
+  }
+}
+
+const replaceRoot = (inputStr, replacements) => {
+  if (!inputStr) return inputStr
+  return inputStr.replace(/\{root\[(.*?)\]\}/g, function (match, p1) {
+    //TODO: fix eslint error
+    //eslint-disable-next-line
+    if (replacements.hasOwnProperty(p1)) {
+      return replacements[p1]
+    }
+    return match
+  })
+}
 
 const WorkfileDetail = ({ workfileId, style }) => {
   const projectName = useSelector((state) => state.project.name)
+  const [selectedSite, setSelectedSite] = useState(null)
```

Comment:
Could we use useLocalStorage here so the user doesn't have to set it everytime? In the future we could use redux global state if siteId is useful to know in other parts of the app.